### PR TITLE
[pre-commit] add more hooks for formatting

### LINKED ISSUE #393: pre-commit hook for clang format
Add pre-commit hook to run clang format, so that all files are properly formatted on commit.

### PR DESCRIPTION
**Pull request by pilkicTT**
Updated the pre-commit config with more checks:
- black - formatter for python files
- clang-format - formatter for c++ files
- generic hooks for detecting trailing whitespace, empty lines on the end of file, etc.

Documentation is also updated, providing instructions on how to setup pre-commit hooks as well as how to build and serve the docs.

NOTE: This change will be followed by the changes for formatting all of the files (since majority of them are not properly formatted) and enabling pre-commit checks to be ran as part of CI.

Closes #393

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,6 +1,24 @@
...
       - id: check-copyright
-        args: ['--config', '.github/check-spdx.yaml']
\ No newline at end of file
+        args: ['--config', '.github/check-spdx.yml']
+  - repo: https://github.com/psf/black
+    rev: 22.3.0
+    hooks:
+      - id: black
+        language_version: python3
+        args: [--line-length=120]
```
Comment (nvukobratTT) [inline .pre-commit-config.yaml:12]: <3

File: docs/src/tools.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Tools
+
+This section will cover setup of various tools that can help you with development of tt-forge-fe.
+
+## Pre-commit
```
Comment (nvukobratTT) [inline docs/src/tools.md:5]: Can we include formatters as pre-commit checks as well (if not already there :))? By [docs](https://pre-commit.com/hooks.html), seems like it's possible to hook them, but didn't try it out. 

@vmilosevic it'll be good to add formaters and pre-commit as part of docker :)) What do you think?

Comment (pilkicTT) [reply docs/src/tools.md:5]: Not sure I follow? The formatters are included as pre-commit checks, see `.pre-commit-config.yaml`...

Comment (pilkicTT) [reply docs/src/tools.md:5]: The `pre-commit` tool takes the config file (`.pre-commit-config.yaml`) and installs appropriate git hooks. End result is that those hooks will be executed whenever you run `git commit`.

Comment (nvukobratTT) [reply docs/src/tools.md:5]: Thanks, that answers my question! 

In sum, I didn't saw that it was a part of the pre-commit config file.. All good! Thanks for the explanation.

**Review by nvukobratTT — APPROVED** [review]

**Comment by pilkicTT** [comment]
I will merge this now to reduce synchronization issues with other commits (since I need to format everything and add a CI job).

**Pull request merged**